packages/bot-web-ui/src/stores/self-exclusion-store.js

```diff
@@ -0,0 +1,74 @@
+import { observable, action, computed } from 'mobx';
+
+export default class SelfExclusionStore {
+    constructor(root_store) {
+        this.root_store = root_store;
+        this.dbot = this.root_store.dbot;
+    }
+
+    @observable max_losses = 0;
+    @observable new_max_losses = 0;
+    @observable run_limit = -1;
+    @observable is_restricted = false;
+
+    @computed
+    get initial_values() {
+        return {
+            new_max_losses: this.new_max_losses || 0,
+            run_limit: this.run_limit,
+        };
+    }
+
+    @action.bound
+    checkClientRestriction() {
+        const { client } = this.root_store.core;
+        // !client.is_eu should update to client.is_eu it's just for test because of restrictions on eu accounts
+        if (!client.is_eu && !client.is_virtual && (!this.max_losses || this.run_limit === -1)) {
+            this.is_restricted = true;
+        } else {
+            this.is_restricted = false;
+        }
+    }
+
+    @action.bound
+    updateMaxLosses(max_losses) {
+        if (max_losses) {
+            this.max_losses = max_losses;
+        }
+    }
+
+    @action.bound
+    updateRunLimit(run_limit) {
+        this.run_limit = run_limit;
+    }
+
+    @action.bound
+    SetNewMaxLosses(new_max_losses) {
+        this.new_max_losses = new_max_losses;
+    }
+
+    @action.bound
+    updateShouldBotRun() {
+        this.should_bot_run = !this.should_bot_run;
+    }
+
+    @action.bound
+    resetSelfExclusion() {
+        this.is_restricted = false;
+        this.max_losses = 0;
+        this.new_max_losses = 0;
+        this.run_limit = -1;
+    }
+
+    @action.bound
+    async checkRestriction() {
+        const { client } = this.root_store.core;
+        const self_exclusion = await client.getSelfExclusion();
+        this.updateMaxLosses(self_exclusion?.get_self_exclusion?.max_losses);
+        this.checkClientRestriction();
+        if (client.self_exclusion.max_losses) {
+            this.updateMaxLosses(client.self_exclusion.max_losses);
```

Comment:
- Can we rename this to `setPreviousMaxLosses` and have "new max losses" be renamed to "max losses"? Currently I'm a bit confused as to which value is the most current.
- We're calling `updateMaxLosses` twice in a row, is that required?

Reply:
code updated and now check changing max_losses in form by using component state